Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/TiersRepository.cs

```diff
@@ -26,6 +26,12 @@ public async Task AddAsync(Tier tier, CancellationToken cancellationToken)
         await _dbContext.SaveChangesAsync(cancellationToken);
     }
 
+    public async Task<bool> ExistsWithName(string tierName, CancellationToken cancellationToken)
+    {
+        var tier = await _tiersDbSet.FirstOrDefaultAsync(t => t.Name == tierName, cancellationToken);
```

Comment:
Always try to transfer as little data as possible to and from the database.

```suggestion
        return await _tiersDbSet.AnyAsync(t => t.Name == tierName, cancellationToken);
```